api/v1/controllers/samples.js

```diff
@@ -197,7 +201,10 @@ module.exports = {
    *  bulk upsert request has been received.
    */
   bulkUpsertSample(req, res/* , next */) {
-    utils.noReadOnlyFieldsInReq(req, helper);
+
+    // make the name post-able
+    const readOnlyFields = helper.readOnlyFields.filter((field) => field !== 'name');
+    utils.noReadOnlyFieldsInReq(req, readOnlyFields);
     const resultObj = { reqStartTime: new Date() };
```

Comment:
@annyhe @iamigo,  for now,  do we want to take out line 207 and merge the PR. This comment was not related to this story. Maybe we can create another story to handle it.